pkg/oc/cli/rollback/rollback.go

```diff
@@ -267,22 +267,13 @@ func (o *RollbackOptions) Run() error {
 		return err
 	}
 
-	// If this is a dry run, print and exit.
-	if o.DryRun {
-		describer := describe.NewDeploymentConfigDescriber(o.appsClient, o.kc, newConfig)
-		description, err := describer.Describe(newConfig.Namespace, newConfig.Name, kprinters.DescriberSettings{})
+	// If an output format is specified or this is a DryRun, print and exit.
```

Comment:
@juanvallejo @soltysh @liggitt @smarterclayton this behavior stands out as big difference to the CLI feel between upstream and us.  I also thing that asking for a printer format does not generally imply that I don't want to perform the actual action.  I think our behavior should change.  Suggestions for a reasonable way to fix it?

Reply:
other than deprecating the implicit "output means dry run" behavior, starting to print a warning now if they set output without dry-run, and making `--dry-run` required in a future release, not really

Reply:
> other than deprecating the implicit "output means dry run" behavior, starting to print a warning now if they set output without dry-run, and making --dry-run required in a future release, not really

Long as we thing that's enough, let's do that.

@mfojtik card, bug, or issue?